### ISSUE TITLE
.Net: FunctionCallingStepwisePlanner doesn't work with OpenAI versioned models

### ISSUE THREAD

**Issue by HillPhelmuth** [microsoft/semantic-kernel#4135]
**Describe the bug**
The current version of GeneratePlan.yaml has `execution_settings` with specific modelIds of `gpt-4` and `gpt-3.5-turbo`. If the kernel used to create the planner has `IChatCompletionService` or `ITextGenerationService` registered with a modelId of `gpt-4-1106-preview` or `gpt-3.5-turbo-1106`, it throws the following error:

```
Unhandled exception. Microsoft.SemanticKernel.KernelException: Required service of type Microsoft.SemanticKernel.AI.TextGeneration.ITextGenerationService not registered. Expected serviceIds: |. Expected modelIds: gpt-4|gpt-3.5-turbo.
   at Microsoft.SemanticKernel.Services.OrderedAIServiceSelector.SelectAIService[T](Kernel kernel, KernelFunction function, KernelArguments arguments)...
```
I haven't tried this with Azure OpenAI. I assume `gpt-4` would work fine there as the model version isn't included in the modelId, however, I would expect to see the same error using `gpt-35-turbo` as the modelId is slightly different.

**To Reproduce**
1. Set up `KernelSyntaxExamples` with any OpenAI chat model other than gpt-4 or gpt-3.5-turbo.
2. Execute `Example66_FunctionCallingStepwisePlanner`
3. See error

**Expected behavior**
Ability to run planner with any available chat completion model.

I'm not sure why the two `modelId`s are included in the `execution_settings` as they are otherwise identical. Removing one of the settings and leaving out the `modelId` should resolve this issue.
```
execution_settings:
  - temperature:       0.0
    top_p:             0.0
    presence_penalty:  0.0
    frequency_penalty: 0.0
    max_tokens:        256
    stop_sequences:    []
```

**Platform**
 - OS: Windows
 - IDE: Visual Studio
 - Language: C#
 - Source: NuGet package version 1.0.0-rc3 or main branch of repository

**Additional context**

**Comment by matthewbolanos** [comment]
@gitri-ms, to make sure there is fallback logic, we need to add a third execution setting that doesn't have a model ID or service ID.

**Comment by gitri-ms** [comment]
Fixed by #4170
